### PR TITLE
Don't try to delete non-existent theme url (bug 1142449)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1142449](https://bugzilla.mozilla.org/show_bug.cgi?id=1142449)

### CODE REVIEW

**Comment by magopian** [comment]
@clouserw @kmaglione r? I'd like to include this in today's push if possible, if someone can QA it once it's merged.

**Comment by diox** [comment]
You've got a typo in the commit message / pull request title :)

File: apps/editors/tests/test_views_themes.py [hunk 1]
```diff
@@ -628,6 +627,41 @@ def test_single_rejected_reason_9_bug_1140346(self):
...
+        # With footer (RereviewQueueTheme created in theme_factory has one).
+        ThemeLock.objects.create(theme=theme, reviewer=user,
+                                 expiry=self.days_ago(-1))
+
+        with self.settings(ALLOW_SELF_REVIEWS=True):
+            res = self.client.post(reject_url, form_data)
+        eq_(res.status_code, 302)
+        assert delete_mock.call_count == 2  # Did try to delete the footer.
+
+        delete_mock.call_count = 0  # Clean slate.
```
Comment (diox) [inline apps/editors/tests/test_views_themes.py]: `delete_mock.reset_mock()` is probably better

File: apps/editors/tests/test_views_themes.py [hunk 2]
```diff
@@ -628,6 +627,41 @@ def test_single_rejected_reason_9_bug_1140346(self):
...
+        # Without footer.
+        RereviewQueueTheme.objects.create(theme=theme,
+                                          header='pending_header.png')
+        ThemeLock.objects.create(theme=theme, reviewer=user,
+                                 expiry=self.days_ago(-1))
+
+        with self.settings(ALLOW_SELF_REVIEWS=True):
+            res = self.client.post(reject_url, form_data)
+        eq_(res.status_code, 302)
+        assert delete_mock.call_count == 1  # Didn't try to delete the footer.
```
Comment (diox) [inline apps/editors/tests/test_views_themes.py]: You should also try to look at the arguments to make sure what was being deleted.

Comment (magopian) [inline apps/editors/tests/test_views_themes.py]: done ;)

**Comment by magopian** [comment]
It's a good thing the branch name was correct, this way, thanks to a stealth `git commit --amend` i could camouflage all this... no one will ever know! Oh, wait...

**Comment by diox** [comment]
r+ once tests are done and they pass

**Pull request merged**